src/addDefaultProps.coffee

```diff
@@ -0,0 +1,21 @@
+import {isFunction} from './util/helpers'
+import addProps from './addProps'
+
+addDefaultProps = (createDefaults) ->
+  addProps (props) ->
+    defaults = if isFunction createDefaults
```

Comment:
Something that's worth thinking about is that technically if the creation function is expensive we probably shouldn't be calling it on every render like this

I checked and eg `addState()` also currently suffers from this

So maybe add a separate issue for going through all the helpers and avoiding calling initializer callbacks more than once (ie just on first render)?

Reply:
That's an issue for `addState`, but for `addDefaultProps`, isn't it necessary to recalculate it every time? It's a function of the current props, and could potentially be different on every render.

Reply:
Ah good point, it seems strange to have the default props be dynamic between renders, but I suppose that's the expected semantics